src/main/java/io/cdap/plugin/http/source/common/ConfigUtil.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Methods for configs validation and configs conversion.
+ */
+public class ConfigUtil {
+  public static void assertIsSet(Object propertyValue, String propertyName, String reason) {
+    if (propertyValue == null) {
+      throw new InvalidConfigPropertyException(
+        String.format("Property '%s' must be set, since %s", propertyName, reason), propertyName);
+    }
+  }
+
+  public static void assertIsNotSet(Object propertyValue, String propertyName, String reason) {
+    if (propertyValue != null) {
+      throw new InvalidConfigPropertyException(
+        String.format("Property '%s' must not be set, since %s", propertyName, reason), propertyName);
+    }
+  }
+
+
+  public static <T extends EnumWithValue> T
+  getEnumValueByString(Class<T> enumClass, String stringValue, String propertyName) {
+    return Stream.of(enumClass.getEnumConstants())
+      .filter(keyType -> keyType.getValue().equalsIgnoreCase(stringValue))
+      .findAny()
+      .orElseThrow(() -> new InvalidConfigPropertyException(
+        String.format("Unsupported '%s' property value: '%s'", propertyName, stringValue), propertyName));
+  }
+
+  @Nullable
+  public static Long toLong(String value, String propertyName) {
+    if (Strings.isNullOrEmpty(value)) {
+      return null;
+    }
+
+    try {
+      return Long.parseLong(value);
+    } catch (NumberFormatException ex) {
+      throw new InvalidConfigPropertyException(String.format("Unsupported value for %s: %s", propertyName, value),
+                                               propertyName);
+    }
+  }
+
+  @Nonnull
```

Comment:
We can leave this out. By convention, anything in our projects that is not annotated as Nullable is non-null.

Reply:
done